lib/server/node-manager.ts

```diff
@@ -28,27 +28,29 @@ export class NodeManager {
     }
 
     public start (callback?: () => void) {
-        let self = this;
+        const self = this;
         this.server = net.createServer(function(socket: Socket) {
             let initialConnection = true;
             let node: Node = null;
-            let socketHandler = new SocketHandler(socket, function(message: string, messageID?: number) {
+            const socketHandler = new SocketHandler(socket, function(socketMessage: SocketMessage) {
                 // We do special handling when we first connect
+                const strMessage: string = socketMessage.asString();
+
                 if (initialConnection) {
                     let connectData: any = null;
                     try {
-                        connectData = JSON.parse(message);
+                        connectData = JSON.parse(strMessage);
```

Comment:
I would encaptulate this on SocketMessage - isJSON() - then it looks like this:
if (!socketMessage.isJSON()) {
     // Exception handling
}

const data = socketMessage.asJSON();